encoder/v1/encoder.go

```diff
@@ -0,0 +1,218 @@
+package v1
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	encoders "github.com/smartcontractkit/ocr2keepers/encoder"
+	"github.com/smartcontractkit/ocr2keepers/pkg/chain"
+	"github.com/smartcontractkit/ocr2keepers/pkg/types"
+)
+
+const (
+	separator = "|"
+)
+
+var (
+	errUpkeepKeyNotParsable = fmt.Errorf("upkeep key not parsable")
+)
+
+type wrappedPerform struct {
+	CheckBlockNumber uint32   `abi:"checkBlockNumber" json:"checkBlockNumber"`
+	CheckBlockhash   [32]byte `abi:"checkBlockhash" json:"checkBlockhash"`
+	PerformData      []byte   `abi:"performData" json:"performData"`
+}
+
+type reportPacker interface {
```

Comment:
Defining this reportPacker interface makes testing easier, since we can mock the behaviour of the pack and unpack operations